core/templates/cowdata.h

```diff
@@ -379,5 +385,7 @@ template <class T>
 CowData<T>::~CowData() {
 	_unref(_ptr);
 }
-
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic pop
+#endif
```

Comment:
```suggestion

#if defined(__GNUC__) && !defined(__clang__)
#pragma GCC diagnostic pop
#endif

```
Let's keep some whitespace to avoid having unrelated stuff crammed together here.